src/entity/specs.ts

```diff
@@ -399,4 +399,74 @@ export const ENTITIES: Record<string, EntityDescriptor<EntitySpec>> = {
       },
     },
   },
+  siSettlement: {
+    label: "Settlement",
+    collectionId: "oracle_collection:sundered_isles/settlement",
+    nameGen: (ent) =>
+      `${(ent.name_cursed && ent.name_cursed.length) ? ent.name_cursed[0].simpleResult : ent.name[0]?.simpleResult}`,
```

Comment:
We don't yet support cursed cursed oracles in entities (pending #349), so you can just have this be `ent.name[0].simpleResult`.